### ISSUE TITLE
Getting Disconnected instead of ConnectionFailure

### ISSUE THREAD

**Issue by francisdb** [fcabestre/Scala-MQTT-client#3]
I used the example code from LocalSubscriber but in connecting state I get a Disconnected instead of ConnectionFailure

**Comment by fcabestre** [comment]
I didn't do anything to enable gitter. So I suppose it's currently disabled. As far as I understand there is not much effort to do to implement ```ExactlyOnce``` from a subscriber's standpoint. So I'd say it's correctly implemented to listen to ```#```.

**Comment by fcabestre** [comment]
I spotted the issue. In my first intents ```ConnectionFailure``` was just aimed at reporting the connection failure as brought by the ```Connack``` frame. At that time I considered the ```TCP``` connection separately from the ```MQTT``` connection. Hence failure at the ```TCP``` level was reported by a distinct API message.

Recently I realised that, from an ```MQTT``` user perspective, both connection processes should be merged. An instead of adding a new reason to the  ```ConnectionFailure``` API message -- for example  ```TCPConnectionFailure```-- I choose to just to say that we are still disconnected. 

This is probably erroneous. What's your opinion on that ? However I'll also fix the examples which are somehow inconsistent.
